app/models/ZuoraCatalog/ZuoraCatalog.scala

```diff
@@ -0,0 +1,72 @@
+package models.ZuoraCatalog
+
+case class ZuoraCatalogPricing(
+    currency: String,
+    price: Double
+)
+
+case class ZuoraCatalogCharge(
+    endDateCondition: Option[String],
+    pricing: List[ZuoraCatalogPricing]
+)
+
+case class ZuoraCatalogProductRatePlan(
+    name: Option[String],
+    id: String,
+    productRatePlanCharges: List[ZuoraCatalogCharge]
+) {
+  def price: Double = {
+    productRatePlanCharges
+      .filter(_.endDateCondition.contains("Subscription_End"))
+      .map(
+        _.pricing
+          .filter(_.currency == "GBP")
+          .map(_.price).sum
+      ).sum
+  }
+}
+
+case class ZuoraCatalogProduct(
+    productRatePlans: List[ZuoraCatalogProductRatePlan]
+)
+
+case class ZuoraCatalog(
+    products: List[ZuoraCatalogProduct]
+)
+
+case class ZuoraCatalogPricePlan(
+    id: String,
+    name: Option[String],
+    pricePerPeriod: List[ZuoraCatalogPricing]
+)
+
+object ZuoraCatalogPricePlan {
+  def build(ZuoraCatalog: ZuoraCatalog, plans: List[String]): List[ZuoraCatalogPricePlan] = {
```

Comment:
this function could be refactored into a for comprehension
you could find an alternative to using `.head`

Reply:
how would you go about doing that?

Reply:
How would you go about doing this?